.circleci/config.yml

```diff
@@ -1,8 +1,171 @@
- version: 2
- jobs:
-   build:
-     docker:
-       - image: circleci/ruby:2.4.1
-     steps:
-       - checkout
-       - run: echo "A first hello"
+defaults: &defaults
+    docker:
+        - image: circleci/node:8.12.0-browsers
+
+version: 2
+jobs:
+    pull_frontend_codebase:
+        <<: *defaults
+        working_directory: /tmp/repos/cbioportal
+        steps:
+            - checkout
+            - run:
+                name: Pull frontend code
+                command: |
+                    export FRONTEND_VERSION=$(grep 'frontend\.version' pom.xml | sed 's/<frontend\.version>//g' | sed 's|</frontend\.version>||' | tr -d '[:blank:]') && \
+                    cd .. && \
+                    git clone https://github.com/$CIRCLE_PROJECT_USERNAME/cbioportal-frontend.git && \
+                    cd cbioportal-frontend && \
+                    git fetch --tags && \
+                    git checkout ${FRONTEND_VERSION}
+            - persist_to_workspace:
+                root: /tmp/repos
+                paths:
+                    - cbioportal-frontend
+
+    install_yarn:
```

Comment:
Once the e2e tests are using the backend of this PR the `install_yarn` job prolly won't be necessary anymore because the frontend artifacts are already in the war. No need to recompile them

Reply:
Hi @inodb Unless I'm misunderstanding something, it is already using the  [backend](https://github.com/cBioPortal/cbioportal/blob/54d2812bc57e99410981564c60ea3d51d6de05c1/.circleci/config.yml#L95).

Reply:
Good point - so it looks like it's pretty much working then 🎉. My point was mostly regarding some minor optimization. I think currently the localdb test tries to mount the frontend inside the war. That's why you need these steps of yarn run build etc. On the frontend e2e tests that makes sense cause you need to test against that frontend code in the PR. On the backend it's redundant b/c the war already contains the compiled frontend. 

Another minor thing is that in the localdb tests the backend code gets pulled from GitHub again and the docker image is made. An optimization would be to create the image before running these tests and using it subsequently in the localdb tests ( to avoid the extra pull). You can then extend it with things like: only rebuild the image when the code has changed.

But yeah we can skip these optimizations for now I think, better to have something working in master branch.

Maybe just double check the backend of the PR is actually being used e.g. by introducing an error or something. And check it is properly creating the new backend image when you add a new commit in the pr to fix it.

Reply:
It is true that the only use of the backend checkout right now is to get the version of the frontend to checkout.  I explored the idea of creating an image from the checkout step done by the pull_frontend_codebase job.  In one scenario, it may just involve the modification of [build_portal_image.sh](https://github.com/cBioPortal/cbioportal-frontend/blob/master/end-to-end-test/local/docker/build_portal_image.sh), maybe with some additional args passed to [setup_docker_containers.sh](https://github.com/cBioPortal/cbioportal-frontend/blob/master/end-to-end-test/local/docker/setup_docker_containers.sh) (called by the circle-ci config).  @pvannierop any thoughts on this?

I did see from the circle-ci page that my hash was being used during the Setup docker images and containers step.  I went ahead and introduced an error as you suggested to confirm the error and fix on revert.  This looks good.

Reply:
@inodb I have a further thought on this - maybe I could add a check to see what files have changed after checkout and then short circuit the rest of the script when code files haven't changed.

Reply:
@inodb based on what I've discovered and communicated to you during our conversation this morning, I'm going to drop the effort to optimize the current configuration.   I'll merge this PR once you give the approval.  cc @pvannierop